Triangulation_3/test/Triangulation_3/test_simplex_iterator_3.cpp

```diff
@@ -0,0 +1,436 @@
+// Copyright (c) 1998  INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+//
+// Author(s)     : Jane Tournois
+
+
+#include <CGAL/Exact_predicates_exact_constructions_kernel.h>
+#include <CGAL/Delaunay_triangulation_3.h>
+
+#include <assert.h>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <vector>
+
+#include <CGAL/IO/read_xyz_points.h>
+#include <CGAL/Random.h>
+
+// Define the kernel.
+typedef CGAL::Exact_predicates_exact_constructions_kernel     Kernel;
+typedef Kernel::Point_3                                       Point_3;
+typedef Kernel::Vector_3                                      Vector_3;
+typedef Kernel::Segment_3                                     Segment_3;
+
+// Define the structure.
+typedef CGAL::Delaunay_triangulation_3< Kernel >    DT;
+
+typedef DT::Vertex_handle                           Vertex_handle;
+typedef DT::Cell_handle                             Cell_handle;
+typedef DT::Edge                                    Edge;
+typedef DT::Facet                                   Facet;
+typedef DT::Segment_simplex_iterator                Segment_simplex_iterator;
+
+void test_vertex_edge_vertex(const DT& dt, const std::size_t& nb_tests)
```

Comment:
shouldn't you have assertions with the number of visited cells? (+per type is possible)

Reply:
It is done in `test_segment_simplex_traverser_3.cpp`. These tests are specifically designed to check that degenerate cases behave as expected